internal/systemd/systemd.go

```diff
@@ -0,0 +1,99 @@
+// Package systemd provides a wrapper around systemd dbus API that allows basic
+// service operations (start/stop/enable/disable).
+package systemd
+
+import (
+	"context"
+	"errors"
+
+	systemdDbus "github.com/coreos/go-systemd/v22/dbus"
+	"github.com/godbus/dbus/v5"
+	"github.com/ubuntu/adsys/internal/i18n"
+	"github.com/ubuntu/decorate"
+)
+
+// Caller is the interface to interact with systemd.
+type Caller interface {
+	StartUnit(context.Context, string) error
+	StopUnit(context.Context, string) error
+
+	EnableUnit(context.Context, string) error
+	DisableUnit(context.Context, string) error
+
+	DaemonReload(context.Context) error
+}
+
+// DefaultCaller is the default implementation of Caller.
+type DefaultCaller struct {
+	conn *systemdDbus.Conn
+}
+
+// jobDone is the string returned by systemd when a job completed successfully.
+const jobDone = "done"
+
+// NewWithConn returns a new systemdCaller using the given dbus connection.
+func NewWithConn(bus *dbus.Conn) (*DefaultCaller, error) {
+	conn, err := systemdDbus.NewConnection(func() (*dbus.Conn, error) { return bus, nil })
+	if err != nil {
+		return nil, err
+	}
+
+	return &DefaultCaller{conn: conn}, nil
+}
+
+// StartUnit starts the given unit.
+func (s *DefaultCaller) StartUnit(ctx context.Context, unit string) (err error) {
```

Comment:
for this one and all the others: there is no change on s itself, right? should that be a copy then and not a pointer?